internal/cephfs/core/metadata.go

```diff
@@ -17,46 +17,97 @@ limitations under the License.
 package core
 
 import (
+	"errors"
 	"fmt"
 	"strings"
+
+	fsAdmin "github.com/ceph/go-ceph/cephfs/admin"
 )
 
 const (
 	// clusterNameKey cluster Key, set on RBD image.
 	clusterNameKey = "csi.ceph.com/cluster/name"
 )
 
+func (s *subVolumeClient) supportsSubVolMetadata(err error) bool {
+	var invalid fsAdmin.NotImplementedError
+	if clusterAdditionalInfo[s.clusterID].subVolMetadataState == unsupported {
+		return false
+	} else if err != nil && errors.Is(err, &invalid) {
```

Comment:
```
[0] pkalever 😎 ceph-csi✨ make
cephcsi image settings: quay.io/cephcsi/cephcsi version canary
if [ ! -d ./vendor ]; then (go mod tidy && go mod vendor); fi
GOOS=linux go build -tags=quincy,ceph_preview,ceph_ci_untested -mod vendor -a -ldflags '-X github.com/ceph/ceph-csi/internal/util.GitCommit=310cbc2025acc7a7703ce67582f1060c13f3d735 -X github.com/ceph/ceph-csi/internal/util.DriverVersion=canary' -o _output/cephcsi ./cmd/
# github.com/ceph/ceph-csi/internal/cephfs/core
internal/cephfs/core/metadata.go:35:41: fsAdmin.NotImplementedError (type) is not an expression
make: *** [Makefile:160: cephcsi] Error 2
[2] pkalever 👿 ceph-csi✨ git diff
diff --git a/internal/cephfs/core/metadata.go b/internal/cephfs/core/metadata.go
index 2dfdcf786..33be13419 100644
--- a/internal/cephfs/core/metadata.go
+++ b/internal/cephfs/core/metadata.go
@@ -30,10 +30,9 @@ const (
 )
 
 func (s *subVolumeClient) supportsSubVolMetadata(err error) bool {
-       var invalid fsAdmin.NotImplementedError
        if clusterAdditionalInfo[s.clusterID].subVolMetadataState == unsupported {
                return false
-       } else if err != nil && errors.Is(err, &invalid) {
+       } else if err != nil && errors.Is(err, fsAdmin.NotImplementedError) {
                // In case the error is other than invalid command return error to the caller.
                clusterAdditionalInfo[s.clusterID].subVolMetadataState = unsupported

```